token/core/fabtoken/actions.go

```diff
@@ -8,13 +8,19 @@ package fabtoken
 import (
 	"encoding/json"
 
-	token2 "github.com/hyperledger-labs/fabric-token-sdk/token/token"
-
 	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
-
 	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/token"
 )
 
+type Signature struct {
+	metadata map[string][]byte // metadata may include for example the preimage of an exchange script
+}
+
+func (s *Signature) Metadata() map[string][]byte {
+	return s.metadata
+}
+
```

Comment:
should this return a copy of the metadata?

Reply:
this is only used in one place and so far there was no problem with using it this way without copying